components/script/dom/htmlinputelement.rs

```diff
@@ -1288,10 +1296,11 @@ impl HTMLInputElement {
             .set_string_attribute(&local_name!("style"), style.into(), can_gc);
     }
 
-    fn update_shadow_tree_if_needed(&self, can_gc: CanGc) {
-        match self.input_type() {
-            InputType::Text => self.update_text_shadow_tree_if_needed(can_gc),
-            InputType::Color => self.update_color_shadow_tree_if_needed(can_gc),
+    fn update_shadow_tree(&self, can_gc: CanGc) {
+        let input_type = self.input_type();
+        match input_type {
+            _ if input_type.is_textual_or_password() => self.update_textual_or_password_shadow_tree(can_gc),
```

Comment:
`is_textual` includes the `hidden` type.

I don't think we should construct a shadow DOM for hidden inputs.